build.py

```diff
@@ -1772,6 +1772,8 @@ def create_docker_build_script(script_name, container_install_dir, container_ci_
             runargs += ["-v", "\\\\.\pipe\docker_engine:\\\\.\pipe\docker_engine"]
         else:
             runargs += ["-v", "/var/run/docker.sock:/var/run/docker.sock"]
+            if os.path.exists(os.path.expanduser("~/.docker/config.json")):
+                runargs += ["-v", os.path.expanduser("~/.docker/config.json:/root/.docker/config.json")]
```

Comment:
I don't really like this change. 
It's kind of we allow to add something without validation. 
We have no control of `~/.docker/config.json` entries there why we may be forced to review some false positive issues related to user environment.

I may need a bit more context, sound like user looking for a proxy config during the build process, and I may be mistaken here but the issue happen when we run Docker-out-of-Docker configuration using `build.py` and user is behind the proxy.

I suppose it's better to add a flag for users `--use-user-docker-config` and add value accordingly if flag is given.

Reply:
I agree a flag may be more appropriate here to be clear on the intent